src/main/java/com/google/step/datamanager/DealManagerDatastore.java

```diff
@@ -105,4 +136,66 @@ public Deal updateDeal(Deal deal) {
     searchManager.putDeal(deal, new ArrayList<>());
     return readDeal(deal.id);
   }
+
+  /** Retrieves deals posted by _ followed by user */
+  private List<Deal> getDealsPublishedByFollowedRestaurantsOrUsers(
```

Comment:
can you split up this into 2 functions `getDealsPublishedByFollowedRestaurants` and `getDealsPublishedByFollowedUsers` ?